### PR TITLE
Add support for tuple accessor methods

### LINKED ISSUE #366: Tuple accessor methods will throw UnsupportedOperationException
https://github.com/eclipse-vertx/vertx-sql-client/blob/3fcbb7cbd533ec14cb3cf0a32ca5f2319f2f2758/vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java#L42-L50

### PR DESCRIPTION
**Pull request by BillyYccc**
Fixes #366

### CODE REVIEW

File: vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java [hunk 1]
```diff
@@ -41,12 +42,31 @@ public ArrayTuple(Collection<?> c) {
 
   @Override
   public <T> T get(Class<T> type, int pos) {
-    throw new UnsupportedOperationException();
+    Object value = this.get(pos);
+    if (value.getClass() == type) {
```
Comment (vietj) [inline vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java]: check nullity and add a test for it

File: vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java [hunk 2]
```diff
@@ -41,12 +42,31 @@ public ArrayTuple(Collection<?> c) {
...
+      }
+      throw new IllegalArgumentException("mismatched type [" + type.getName() + "] for the value of type [" + value.getClass().getName() + "]");
+    }
   }
 
   @Override
   public <T> T[] getValues(Class<T> type, int pos) {
-    throw new UnsupportedOperationException();
+    Object value = this.get(pos);
+    if (value.getClass().isArray() && value.getClass().getComponentType() == type) {
```
Comment (vietj) [inline vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java]: add nullity check and test for null

File: vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java (hunk 1 above)
Comment (BillyYccc) [reply vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ArrayTuple.java]: added in https://github.com/eclipse-vertx/vertx-sql-client/pull/367/commits/4beca16bc56cc291c3e8f0b73d3c9517dc3c5055.
